packages/sitecore-jss/src/layout/rest-layout-service.ts

```diff
@@ -188,58 +191,44 @@ export class RestLayoutService extends LayoutServiceBase {
   }
 
   /**
-   * Provides default @see AxiosDataFetcher data fetcher
+   * Returns a fetcher function pre-configured with headers from the incoming request.
+   * Provides default @see NativeDataFetcher data fetcher
    * @param {IncomingMessage} [req] Request instance
-   * @param {ServerResponse} [res] Response instance
    * @returns default fetcher
    */
-  protected getDefaultFetcher = <T>(req?: IncomingMessage, res?: ServerResponse) => {
-    const config = {
-      debugger: debug.layout,
-    } as AxiosDataFetcherConfig;
-    if (req && res) {
-      config.onReq = this.setupReqHeaders(req);
-      config.onRes = this.setupResHeaders(res);
-    }
-    const axiosFetcher = new AxiosDataFetcher(config);
+  protected getDefaultFetcher = <T>(req?: IncomingMessage) => {
+    const config: NativeDataFetcherConfig = { debugger: debug.layout };
 
-    const fetcher = (url: string, data?: unknown) => {
-      return axiosFetcher.fetch<T>(url, data);
-    };
+    const headers = this.getHeaders(req);
+
+    const nativeFetcher = new NativeDataFetcher(config);
 
-    return fetcher;
+    return (url: string, data?: RequestInit) => nativeFetcher.fetch<T>(url, { ...data, headers });
   };
 
   /**
-   * Setup request headers
-   * @param {IncomingMessage} req Request instance
-   * @returns {AxiosRequestConfig} axios request config
+   * Creates an HTTP `Headers` object populated with headers from the incoming request.
+   * @param {IncomingMessage} [req] - The incoming HTTP request, used to extract headers.
+   * @returns {Headers} - An instance of the `Headers` object populated with the extracted headers.
    */
-  protected setupReqHeaders(req: IncomingMessage) {
-    return (reqConfig: AxiosRequestConfig) => {
-      debug.layout('performing request header passing');
-      reqConfig.headers.common = {
-        ...reqConfig.headers.common,
-        ...(req.headers.cookie && { cookie: req.headers.cookie }),
-        ...(req.headers.referer && { referer: req.headers.referer }),
-        ...(req.headers['user-agent'] && { 'user-agent': req.headers['user-agent'] }),
-        ...(req.connection.remoteAddress && { 'X-Forwarded-For': req.connection.remoteAddress }),
-      };
-      return reqConfig;
-    };
-  }
+  private getHeaders(req?: IncomingMessage): Headers {
+    const headers = new Headers();
+
+    if (req?.headers) {
+      // Copy all headers from req.headers
+      Object.entries(req.headers).forEach(([key, value]) => {
+        if (value) {
+          headers.set(key, Array.isArray(value) ? value.join(', ') : value);
+        }
+      });
+
+      // Add or override specific headers
+      req.headers.cookie && headers.set('cookie', req.headers.cookie);
```

Comment:
Is there a need for this section? except for `X-Forwarded-For`, every header should have been set on lines 219-221.

Reply:
i agree but even if these were copied earlier, overriding ensures their presence.